test/e2e/autoscale_test.go

```diff
@@ -55,69 +55,22 @@ func isDeploymentScaledToZero() func(d *v1beta1.Deployment) (bool, error) {
 	}
 }
 
-func generateTraffic(clients *test.Clients, logger *logging.BaseLogger, concurrency int, duration time.Duration, domain string) error {
-	var (
-		totalRequests      int
-		successfulRequests int
-		mux                sync.Mutex
-		group              errgroup.Group
-	)
-
-	logger.Infof("Maintaining %d concurrent requests for %v.", concurrency, duration)
-	for i := 0; i < concurrency; i++ {
-		group.Go(func() error {
-			done := time.After(duration)
-			client, err := pkgTest.NewSpoofingClient(clients.KubeClient, logger, domain, test.ServingFlags.ResolvableDomain)
-			if err != nil {
-				return fmt.Errorf("error creating spoofing client: %v", err)
-			}
-			req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("http://%s", domain), nil)
-			if err != nil {
-				return fmt.Errorf("error creating spoofing client: %v", err)
-			}
-			for {
-				select {
-				case <-done:
-					return nil
-				default:
-					mux.Lock()
-					totalRequests++
-					mux.Unlock()
-					res, err := client.Do(req)
-					if err != nil {
-						logger.Errorf("error making request %v", err)
-						continue
-					}
-
-					if res.StatusCode != http.StatusOK {
-						logger.Errorf("non 200 response %v", res.StatusCode)
-						logger.Errorf("response headers: %v", res.Header)
-						logger.Errorf("response body: %v", res.Body)
-						continue
-					}
-					mux.Lock()
-					successfulRequests++
-					mux.Unlock()
-				}
-			}
-		})
-	}
-
-	logger.Infof("Waiting for all requests to complete.")
-	if err := group.Wait(); err != nil {
-		return fmt.Errorf("Error making requests for scale up: %v.", err)
-	}
-
-	if successfulRequests != totalRequests {
-		return fmt.Errorf("Error making requests for scale up. Got %d successful requests. Wanted %d.",
-			successfulRequests, totalRequests)
-	}
+func tearDown(ctx *testContext) {
+	TearDown(ctx.clients, ctx.names, ctx.logger)
+}
 
-	logger.Infof("All %d requests succeeded.", totalRequests)
-	return nil
+type testContext struct {
+	t              *testing.T
```

Comment:
Not a huge fan of having `testing.T` inside the testContext… Especially as it doesn't seem to be used in the `assert` function (doing a `logger.Fatal` instead)

Reply:
I should be doing `ctx.t.Fatalf` instead to fail the test.